drivers/adc/adc_mcux_adc16.c

```diff
@@ -61,6 +89,15 @@ static int mcux_adc16_channel_setup(struct device *dev,
 		return -EINVAL;
 	}
 
+#ifdef CONFIG_ADC_HW_TRIGGER
+	if (CONFIG_ADC_HW_TRIGGER) {
+		/* enable ADC trigger channel */
+		SIM->SOPT7 |=
+			SIM_SOPT7_ADC0TRGSEL(CONFIG_ADC_HW_TRIGGER_CHANNEL) |
+			SIM_SOPT7_ADC0ALTTRGEN(1);
```

Comment:
SOPT registers are usually soc-specific. Will this work on multiple platforms?

Reply:
I checked for kenetis platfroms with adc16 seems they are all configured to use SOPT7, but yes, I still need think about a better soc dependant way, do you have any recommendation solution. this code really looks uncomfortable.